tephra-core/src/main/java/co/cask/tephra/distributed/TransactionServiceThriftClient.java

```diff
@@ -79,67 +84,141 @@ public void close() {
   }
 
   public Transaction startLong() throws TException {
-    return TransactionConverterUtils.unwrap(client.startLong());
+    try {
+      return TransactionConverterUtils.unwrap(client.startLong());
+    } catch (Exception e) {
+      isValid = false;
+      throw e;
+    }
   }
 
   public Transaction startShort() throws TException {
+    try {
       return TransactionConverterUtils.unwrap(client.startShort());
+    } catch (Exception e) {
+      isValid = false;
+      throw e;
+    }
   }
 
   public Transaction startShort(int timeout) throws TException {
+    try {
       return TransactionConverterUtils.unwrap(client.startShortTimeout(timeout));
+    } catch (Exception e) {
+      isValid = false;
+      throw e;
+    }
   }
 
   public boolean canCommit(Transaction tx, Collection<byte[]> changeIds) throws TException {
-
+    try {
       return client.canCommitTx(TransactionConverterUtils.wrap(tx),
                                 ImmutableSet.copyOf(Iterables.transform(changeIds, BYTES_WRAPPER))).isValue();
+    } catch (Exception e) {
+      isValid = false;
+      throw e;
+    }
   }
 
   public boolean commit(Transaction tx) throws TException {
-
+    try {
       return client.commitTx(TransactionConverterUtils.wrap(tx)).isValue();
+    } catch (Exception e) {
```

Comment:
This would close the connection when a service exception like `TransactionNotInProgressException` is thrown, or some other exception gets thrown during wrap/unwrap, right?